packages/ui-extensions/docs/surfaces/point-of-sale/reference/examples/scanner-api/hardware-scanner-example.tsx

```diff
@@ -0,0 +1,39 @@
+import React, {useEffect} from 'react';
+import {
+  Navigator,
+  Screen,
+  Stack,
+  Text,
+  useScannerDataSubscription,
+  useApi,
+  reactExtension,
+} from '@shopify/ui-extensions-react/point-of-sale';
+
+const SmartGridModal = () => {
+  const api = useApi<'pos.home.modal.render'>();
+  const {data} = useScannerDataSubscription();
+
+  // Update the cart when a product is scanned.
+  useEffect(() => {
+    const addProductToCart = (data: string | undefined) => {
+      if (data) {
+        api.cart.addLineItem(Number(data), 1);
+      }
+    };
+    addProductToCart(data);
+  }, [data]);
```

Comment:
Doesn't shopify assign productID's regardless of what the scanned data is? In other words, does this logic actually make sense?

Reply:
Hm I need more clarification on this. I don't understand

Reply:
The example says "whatever barcode is scanned, add the product with that productID to the cart"

I'm wondering if that example is misleading because a barcode doesn't necessarily match a productID. I think a more accurate example would be to just put the text on the screen somehow.

Reply:
It's also not async so I don't think there's a reason to do the whole "make a function call a function" thing

Reply:
Right! My point is barcode !== productId, so trying to add the barcode as a productId will never match... Here's a pic of admin to illustrate
<img width="911" alt="image" src="https://github.com/Shopify/ui-extensions/assets/29490857/deb11d3a-99c0-494d-9a40-05b8a9b5b88c">

Reply:
Makes sense, thank you for helping me understand! This logic is unnecessary I'll remove it